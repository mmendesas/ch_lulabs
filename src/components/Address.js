import React from "react";

const Address = ({ logradouro, bairro, localidade, uf, cep }) => (
  <section className="address">
    <h1 className="address__title">{logradouro}</h1>
    <div className="address__item">{bairro}</div>
    <p className="address__item">{`${localidade}-${uf}`}</p>
    <div className="address__item">{cep}</div>
  </section>
);

export default Address;
