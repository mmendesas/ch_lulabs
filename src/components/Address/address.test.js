import React from 'react';
import { render } from 'react-testing-library';
import Address from './index';

const setup = () => {
  const info = {
    cep: '79065-260',
    logradouro: 'Rua AmapÃ¡ Doce',
    bairro: 'Vila Moreninha III',
    localidade: 'Campo Grande',
    uf: 'MS'
  };

  return {
    info
  };
};

describe('[Component] - Address', () => {
  it('should be match snapshot', () => {
    const { getByText } = render(<Address />);
    expect(getByText('address')).toMatchSnapshot();
  });

  it('renders address with default values', () => {
    const { getByText } = render(<Address />);
    expect(getByText('address')).toBeInTheDocument();
  });

  it('renders address with properties', () => {
    const { info } = setup();
    const { getByText } = render(<Address {...info} />);
    const { localidade, uf } = info;

    Object.values(info).forEach(prop => {
      const text = [localidade, uf].includes(prop)
        ? `${localidade}-${uf}`
        : prop;
      expect(getByText(text)).toBeInTheDocument();
    });
  });
});
