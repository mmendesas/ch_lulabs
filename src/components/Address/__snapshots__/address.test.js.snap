// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component] - Address should be match snapshot 1`] = `
<h1
  class="address__title"
>
  address
</h1>
`;
